src/hotspot/share/opto/library_call.cpp

```diff
@@ -5502,29 +5502,44 @@ bool LibraryCallKit::inline_array_partition() {
 
   Node* pivotIndices = nullptr;
 
+  const TypeInstPtr* elem_klass = gvn().type(elementType)->isa_instptr();
+  if (elem_klass == nullptr) {
+    return false;  // dead path (elementType->is_top()).
+  }
+  if (obj == nullptr || obj->is_top()) {
+    return false;  // dead path
+  }
+  // java_mirror_type() returns non-null for compile-time Class constants only.
+  ciType* elem_type = elem_klass->java_mirror_type();
+  if (elem_type == nullptr) {
+    return false;
+  }
+  BasicType bt = elem_type->basic_type();
+  // Disable the intrinsic if the CPU does not support SIMD sort
+  if (!Matcher::supports_simd_sort(bt)) {
+    return false;
+  }
+  address stubAddr = nullptr;
+  stubAddr = StubRoutines::select_array_partition_function();
+  // stub not loaded
+  if (stubAddr == nullptr) {
+    return false;
+  }
+  // get the address of the array
+  const TypeAryPtr* obj_t = _gvn.type(obj)->isa_aryptr();
+  if (obj_t == nullptr || obj_t->elem() == Type::BOTTOM ) {
+    return false; // failed input validation
+  }
+
+  null_check(obj);
+  // If obj is dead, only null-path is taken.
+  if (stopped())  return true;
```

Comment:
Can you add braces here?
```suggestion
  if (stopped()) {
    return true;
  }
```

Reply:
done